cython/core/triangulation.pyx

```diff
@@ -2559,18 +2564,9 @@ cdef class Triangulation():
                                           degree,
                                           strategy=strategy,
                                           num_threads=num_threads)
-        return [self.cover(H) for H in reps if len(H[0]) == degree]
-
-    def _covers_low_index_old(self, degree):
-        G = self.fundamental_group()
-
-        if G.num_relators() > self.num_cusps():
-            S = low_index.SimsTree(G.num_generators(), degree, G.relators(),
-                        num_long_relators=self.num_cusps())
-        else:
-            S = low_index.SimsTree(G.num_generators(), degree, G.relators())
-        return [self.cover(H.permutation_rep()) for H in S.list()
-                if H.degree == degree]
+        return [self.cover(H) for H in reps if
+                ((len(H) > 1) and len(H[0]) == degree) or
+                (len(H) == 0 and degree == 1)]
```

Comment:
I am not 100% sure this is correct and find it somewhat confusing?

It might be more readable if this is a conditional on the number of generators.

Something like
`
if degree == 1:
    return self
if G.num_generators() == 0:
    return []
return [ self.cover(H) for H in reps if len(H[0]) == degree ]
`

Reply:
I tested it pretty extensively, but agree it's confusing.  I'll try to make it clearer.

Reply:
I am still not sure whether this is correct if the group has one generator?

Reply:
Also can be expressed in terms of a helper:
```
def degree_of_rep(H):
    if len(H) == 0:
        return 1
    return len(H[0])
```

Reply:
What do we do if the user sets degree = 0?

Reply:
> What do we do if the user sets degree = 0?

Currently, we return the empty list.  If the degree is negative, it barfs up an unhelpful exception.  I'll improve that.

Reply:
> I am still not sure whether this is correct if the group has one generator?

One generator is fine --- no gens is the only corner case:
```
>>> U = snappy.Link([[0, 0, 1, 1]]).exterior()
>>> U.dehn_fill((15, 1))
>>>  U.fundamental_group()
Generators:
   a
Relators:
   aaaaaaaaaaaaaaa
>>> [C.homology() for C in U.covers(5)]
[Z/3]
```

Reply:
```
>>> U.fundamental_group(False)
Generators:
   a,b
Relators:
   AaBabB
   bBabBbBabBabBabBabBabBabBabBabBabBabBabBabBabBa
```

so this doesn't really test the one generator case.

Reply:
> so this doesn't really test the one generator case.

The all the cover code uses the most simplified fundamental group possible, i.e. `U.fundamental_group()` which is equivalent to `U.fundamental_group(True, True, True, True)`.

Reply:
More accurately, the code for **finding** the covers uses the simplified fundamental group.  The kernel code for building the covers uses the original generators constructed by `choose_generators`.  But that's dealt with in `Triangulation.build_rep_into_Sn` by the kernel function `convert_candidateSn_to_original_generators`.

Reply:
Cleaned up code in 3704f845 using helper function as suggested.